### ISSUE TITLE
Unhandled Exception with Static Libs (Apple M1)

### ISSUE THREAD

**Issue by godrays** [SFML/SFML#2453]
Hello,

I built SMFL 2.6.x from source with static library option and compiled a simple application. I do get the following issue when I run the simple application. I also tried SFML release 2.5.1 version but I got the same issue.

```
2023-03-15 18:59:18.928 GameApp[4579:43097] -[SFOpenGLView enableKeyRepeat]: unrecognized selector sent to instance 0x12a75baa0
2023-03-15 18:59:18.929 GameApp[4579:43097] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[SFOpenGLView enableKeyRepeat]: unrecognized selector sent to instance 0x12a75baa0'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000192a243e8 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000019256eea8 objc_exception_throw + 60
	2   CoreFoundation                      0x0000000192ac6c0c -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x000000019298a660 ___forwarding___ + 1600
	4   CoreFoundation                      0x0000000192989f60 _CF_forwarding_prep_0 + 96
	5   GameApp                             0x0000000100dabb4c -[SFOpenGLView initWithFrame:fullscreen:highDpi:] + 84
	6   GameApp                             0x0000000100dacb6c -[SFWindowController setupWindowWithMode:andStyle:] + 188
	7   GameApp                             0x0000000100dac708 -[SFWindowController initWithMode:andStyle:] + 336
	8   GameApp                             0x0000000100dae270 _ZN2sf4priv15WindowImplCocoaC2ENS_9VideoModeERKNS_6StringEmRKNS_15ContextSettingsE + 100
	9   GameApp                             0x0000000100da2034 _ZN2sf4priv10WindowImpl6createENS_9VideoModeERKNS_6StringEjRKNS_15ContextSettingsE + 76
	10  GameApp                             0x0000000100da0bd8 _ZN2sf6Window6createENS_9VideoModeERKNS_6StringEjRKNS_15ContextSettingsE + 400
	11  GameApp                             0x0000000100d8a610 _ZN2sf12RenderWindowC1ENS_9VideoModeERKNS_6StringEjRKNS_15ContextSettingsE + 116
	12  GameApp                             0x0000000100d471bc main + 120
	13  dyld                                0x000000019259fe50 start + 2544
)
libc++abi: terminating with uncaught exception of type NSException

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

Environment

* macOS Ventura 13.2.1 
* SFML 2.6.x branch
* Xcode / Apple clang version 14.0.0 (clang-1400.0.29.202)
  Target: arm64-apple-darwin22.3.0
  Thread model: posix
  InstalledDir: /Library/Developer/CommandLineTools/usr/bin

I have used the following script to build the SFML

```bash
#!/bin/bash

lib_name=sfml
lib_version=2.6.x
lib_url=https://github.com/SFML/SFML.git

pushd .

rm -rf $lib_name

git clone --recurse-submodules -b $lib_version $lib_url ./$lib_name/$lib_version
cd $lib_name/$lib_version

rm -rf build
rm -rf installed
mkdir build
cd build

cmake .. -DCMAKE_BUILD_TYPE=Release -DSFML_BUILD_FRAMEWORKS=FALSE -DBUILD_SHARED_LIBS=FALSE -DCMAKE_INSTALL_PREFIX=../installed
cmake --build . --target install -- -j `nproc --all`

cd ..
rm -rf build

popd
```

Here how I link the libraries:

```
    target_link_libraries(GameApp
            "-framework Carbon -framework CoreFoundation -framework OpenGL -framework Cocoa -framework IOKit"
            "-framework AudioUnit"
            sfml-audio-s
            sfml-graphics-s
            sfml-network-s
            sfml-system-s
            sfml-window-s
            )
```

The simple source code I used:

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode({1280, 720}), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        for (sf::Event event; window.pollEvent(event);)
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

Any thoughts?

Thank you for your help!

**Comment by ChrisThrasher** [comment]
I can't reproduce this. First of all, linking to all of those libraries is unnecessary and error prone. Please link to SFML's CMake targets to ensure that there are no linker issues to blame for this problem. Second, this line:

```cpp
sf::RenderWindow window(sf::VideoMode({1280, 720}), "Minimal, complete and verifiable example");
```

is using SFML 3's interface for `sf::VideoMode`, not SFML 2's. Somehow you're compiling against SFML 3 headers. If you mix those with SFML 2 libraries then there are no promises that anything will work.

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML/pull/1485

Looks like you ran into this issue which I believe has been fixed on both `master` and `2.6.x` for a while now.

**Comment by godrays** [comment]
Thank you @ChrisThrasher. 

Including "-ObjC" in target_link_libraries() fixed the issue.

For reference, the fix is reported here: https://developer.apple.com/library/archive/qa/qa1490/_index.html
